rr_iarrc/src/finish_line_watcher/finish_line_watcher.cpp

```diff
@@ -78,23 +87,102 @@ void ImageCB(const sensor_msgs::ImageConstPtr& msg) {
     frame = cv_ptr->image;
     blockEnvironment(frame);
 
-    auto count = countNonZero(frame);
+    Mat src = frame.clone();
+    Mat dst;
+    cv::Canny(src, dst, canny_thresh1, canny_thresh2, 3);
```

Comment:
It will barely affect performance, but Canny is overkill when you already have a binary image. A simple laplacian is a better way to get the lines.